scripts/cxg_admin.py

```diff
@@ -250,6 +250,49 @@ def migrate_schema_version(ctx):
             record.schema_version = version
 
 
+@cli.command()
+@click.pass_context
+def migrate_dataset_published_on(ctx):
+    """
+    Populates `published_on` for each existing dataset. This is a one-off procedure since new 
+    datasets will have the published_on set when they are added to a collection or the parent
+    collection is published.
+    """
+
+    with db_session_manager() as session:
+        click.confirm(
+            f"Are you sure you want to run this script? It will assign published_on to all the "
+            f"datasets",
+            abort=True,
+        )
+        for record in session.query(DbDataset):
+            dataset_id = record.id
+            dataset_created_at = record.created_at
+            record.published_on = dataset_created_at
+            logger.info(f"Setting published_on for dataset {dataset_id}")
```

Comment:
If you move this at the beginning of the function, in case of error you will know what dataset is affected. Note that this might work regardless because of how commit works, but I think it's better to have a predictable behavior.